controllers/articles.js

```diff
@@ -102,4 +108,126 @@ export default class ArticleController {
       });
     }
   }
+
+  /**
+   * @description - Search for articles
+   * @static
+   * @param {Object} req - the request object
+   * @param {Object} res - the response object
+   * @memberof ArticleController
+   * @returns {Object} class instance
+   */
+  static async searchForArticles(req, res) {
+    try {
+      let { offset, limit } = req.query;
+      offset = Number(offset) || 0;
+      limit = Number(limit) || 10;
+      const searchTerms = ArticleController.generateSearchQuery(req.query);
+      const results = await Article.findAndCountAll({
+        where: {
+          ...searchTerms,
+        },
+        include: [{
+          model: User,
+          attributes: ['username', 'email', 'name', 'bio'],
+          as: 'author',
+        }],
+        offset,
+        limit,
+      });
+      const { count } = results;
+      results.totalPages = Math.ceil(count / limit);
```

Comment:
@leksyib your pagination function should be a helper function. It can be used in multiple places. You pass the necessary parameters and it calculates and returns the required metadata.
If we decide to implement a paginated search for users tomorrow, we should not have to write the logic all over again

Reply:
@9jaswag  I just did that